bazel/repositories.bzl

```diff
@@ -171,6 +183,11 @@ def envoy_dependencies(path = "@envoy_deps//", skip_targets = []):
                 actual = path + ":" + t,
             )
 
+    # Treat Envoy's overall build config as an external repo, so projects that
+    # build Envoy as a subcomponent can easily override the config.
+    if "envoy_build_config" not in native.existing_rules().keys():
```

Comment:
We should add a CI test with a custom config (turning off one component to verify this flow). Alternatively, this can live in envoy-filter-example.

Reply:
Yes will sort this out.